ivy/functional/backends/torch/activations.py

```diff
@@ -107,23 +107,26 @@ def softsign(x: torch.Tensor, /, out: Optional[torch.Tensor] = None) -> torch.Te
 
 
 @with_unsupported_dtypes(
-    {
-        "2.0.1 and below": (
-            "complex",
-            "float16",
-        )
-    },
+    {"2.0.1 and below": ("float16",)},
     backend_version,
 )
 def log_softmax(
     x: torch.Tensor,
     /,
     *,
     axis: Optional[int] = None,
+    complex_mode: Literal["split", "magnitude", "jax"] = "jax",
     out: Optional[torch.Tensor] = None,
 ):
     if axis is None:
         axis = -1
+    if torch.is_complex(x):
+        x_max = torch_backend.max(x, axis=axis, keepdims=True)
+        sub_temp = torch.sub(x, x_max)
+        ret = torch.sum(sub_temp.exp(), dim=axis, keepdim=True)
+        ret = torch.log(ret)
+        ret = torch.sub(sub_temp, ret)
+        return ret
```

Comment:
Since softmax is already implemented why not use `torch.log(torch_backend.softmax(x))`?

Reply:
at first I thought about this approach but when I considered torch documentation about **log_softmax**  that doing **log** and **softmax* ops is computationally expensive also not numerically stable I though of a second implementation which is faster, simpler and also mathematically equivalent. here
log_softmax(x) =log(exp(x)/ sum(exp(x)) => x - log(sum(exp(x))

Reply:
![image](https://github.com/unifyai/ivy/assets/97109954/3e451607-0274-4077-a94d-0c8255c56333)
Here is the torch doc

Reply:
Oh cool, that looks good then